app/src/main/java/com/bashkevich/androidfundamentals/MainActivity.kt

```diff
@@ -1,38 +1,83 @@
 package com.bashkevich.androidfundamentals
 
+import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import androidx.fragment.app.add
-import androidx.fragment.app.commit
+import android.util.Log
+import androidx.fragment.app.*
 import androidx.lifecycle.lifecycleScope
 import androidx.work.*
+import com.bashkevich.androidfundamentals.model.database.MoviesDatabase
+import com.bashkevich.androidfundamentals.model.database.entity.MovieEntity
 import com.bashkevich.androidfundamentals.model.workmanager.MovieLoaderWorker
+import com.bashkevich.androidfundamentals.model.workmanager.MovieNotificationsWorker
+import com.bashkevich.androidfundamentals.model.workmanager.WorkerRepository
+import com.bashkevich.androidfundamentals.moviesdetails.view.FragmentMoviesDetails
 import com.bashkevich.androidfundamentals.movieslist.view.FragmentMoviesList
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import java.util.concurrent.TimeUnit
 
 class MainActivity : AppCompatActivity() {
+    private val workerRepository = WorkerRepository()
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
         if (savedInstanceState == null) {
             supportFragmentManager.commit {
-                setReorderingAllowed(true)
-                add<FragmentMoviesList>(R.id.main_container)
+                replace(R.id.main_container, FragmentMoviesList())
+            }
+            intent?.let(::handleMovieIntent)
+
+            lifecycleScope.launch {
+
+                WorkManager.getInstance(applicationContext).enqueueUniquePeriodicWork(
+                    MovieLoaderWorker.UPDATE_MOVIES,
+                    ExistingPeriodicWorkPolicy.REPLACE,
+                    workerRepository.periodicMovieRequest
+                )
+                WorkManager.getInstance(applicationContext).enqueueUniquePeriodicWork(
+                    MovieNotificationsWorker.NOTIFY,
+                    ExistingPeriodicWorkPolicy.REPLACE,
+                    workerRepository.periodicNotificationRequest
+                )
             }
         }
+    }
 
-        lifecycleScope.launchWhenCreated {
-            val constraints =
-                Constraints.Builder().setRequiredNetworkType(NetworkType.UNMETERED).build()
+    override fun onNewIntent(intent: Intent?) {
```

Comment:
Can intent be null?

Reply:
It's the default function, so I think yes

Reply:
Actually, not. This function will not be called without intent.